src/js/_enqueues/admin/site-icon.js

```diff
@@ -1,138 +1,180 @@
+					suggestedHeight: $el.data( 'size' )
+				}),
+				new wp.media.controller.SiteIconCropper({
+					control: {
+						params: {
+							width: $el.data( 'size' ),
+							height: $el.data( 'size' )
+						}
+					},
+					imgSelectOptions: calculateImageSelectOptions
+				})
+			]
+		});
 
-			// Create the media frame.
-			frame = wp.media({
-				button: {
-					// Set the text of the button.
-					text: $el.data('update'),
-					// Don't close, we might need to crop.
-					close: false
-				},
-				states: [
-					new wp.media.controller.Library({
-						title: $el.data( 'choose' ),
-						library: wp.media.query({ type: 'image' }),
-						date: false,
-						suggestedWidth: $el.data( 'size' ),
-						suggestedHeight: $el.data( 'size' )
-					}),
-					new wp.media.controller.SiteIconCropper({
-						control: {
-							params: {
-								width: $el.data( 'size' ),
-								height: $el.data( 'size' )
-							}
-						},
-						imgSelectOptions: calculateImageSelectOptions
-					})
-				]
-			});
+		frame.on( 'cropped', function( attachment ) {
+			$( '#site_icon_hidden_field' ).val( attachment.id );
+			switchToUpdate( attachment.url );
+			frame.close();
+			// Start over with a frame that is so fresh and so clean clean.
+			frame = null;
+		});
 
-			frame.on( 'cropped', function( attachment) {
-				$( '#site_icon_hidden_field' ).val(attachment.id);
-				switchToUpdate(attachment.url);
+		// When an image is selected, run a callback.
+		frame.on( 'select', function() {
+			// Grab the selected attachment.
+			var attachment = frame.state().get( 'selection' ).first();
+
+			if (
+				attachment.attributes.height === $el.data( 'size' ) &&
+				$el.data( 'size' ) === attachment.attributes.width
+			) {
+				// Set the value of the hidden input to the attachment id.
+				$( '#site_icon_hidden_field' ).val( attachment.id );
+				switchToUpdate( attachment.attributes.url );
 				frame.close();
-				// Start over with a frame that is so fresh and so clean clean.
-				frame = null;
-			});
-
-			// When an image is selected, run a callback.
-			frame.on( 'select', function() {
-				// Grab the selected attachment.
-				var attachment = frame.state().get('selection').first();
-
-				if ( attachment.attributes.height === $el.data('size') && $el.data('size') === attachment.attributes.width ) {
-					// Set the value of the hidden input to the attachment id.
-					$( '#site_icon_hidden_field').val(attachment.id);
-					switchToUpdate(attachment.attributes.url);
-					frame.close();
-				} else {
-					frame.setState( 'cropper' );
-				}
-			});
-
-			frame.open();
+			} else {
+				frame.setState( 'cropper' );
+			}
 		});
-	});
 
-	function switchToUpdate( url ){
-		// Set site-icon-img src to the url and remove the hidden class.
-		$( '#site-icon-preview').find('img').not('.browser-preview').each( function(i, img ){
-			$(img).attr('src', url );
+		frame.open();
+	});
+})( jQuery );
+
+/**
+ * Updates the site icon preview with the specified URL.
```

Comment:
This skips some of the changes that are taking place. Mabe a short description along the lines of "Updates Site Icon settings after a selection is made" and then this along with an overview of the other changes?